impeller/entity/contents/color_source_contents.h

```diff
@@ -100,6 +101,104 @@ class ColorSourceContents : public Contents {
+
+  template <typename VertexShaderT>
+  bool DrawGeometry(GeometryResult geometry_result,
+                    const ContentContext& renderer,
+                    const Entity& entity,
+                    RenderPass& pass,
+                    const PipelineBuilderCallback& pipeline_callback,
+                    typename VertexShaderT::FrameInfo frame_info,
+                    const BindFragmentCallback& bind_fragment_callback) const {
+    auto options = OptionsFromPassAndEntity(pass, entity);
+
+    // If overdraw prevention is enabled (like when drawing stroke paths), we
+    // increment the stencil buffer as we draw, preventing overlapping fragments
+    // from drawing. Afterwards, we need to append another draw call to clean up
+    // the stencil buffer (happens below in this method).
+    if (geometry_result.prevent_overdraw) {
+      options.stencil_mode =
+          ContentContextOptions::StencilMode::kLegacyClipIncrement;
+    }
+    options.primitive_type = geometry_result.type;
+    pass.SetVertexBuffer(std::move(geometry_result.vertex_buffer));
+    pass.SetStencilReference(entity.GetClipDepth());
+
+    // Take the pre-populated vertex shader uniform struct and set managed
+    // values.
+    frame_info.depth = entity.GetShaderClipDepth();
+    frame_info.mvp = geometry_result.transform;
+
+    VertexShaderT::BindFrameInfo(
+        pass, renderer.GetTransientsBuffer().EmplaceUniform(frame_info));
+
+    std::shared_ptr<Pipeline<PipelineDescriptor>> pipeline =
+        pipeline_callback(options);
+    pass.SetPipeline(pipeline);
+
+    // The reason we need to have a callback mechanism here is that this routine
+    // may insert draw calls before the main draw call below. For example, for
+    // sufficiently complex paths we may opt to use stencil-then-cover to avoid
+    // tessellation.
+    if (!bind_fragment_callback(pass)) {
+      return false;
+    }
+
+    if (!pass.Draw().ok()) {
+      return false;
+    }
+
+    // If we performed overdraw prevention, a subsection of the clip heightmap
+    // was incremented by 1 in order to self-clip. So simply append a clip
+    // restore to clean it up.
+    if (geometry_result.prevent_overdraw) {
+      auto restore = ClipRestoreContents();
+      restore.SetRestoreCoverage(GetCoverage(entity));
+      return restore.Render(renderer, entity, pass);
+    }
+    return true;
+  }
+
+  template <typename VertexShaderT>
+  bool DrawPositions(const ContentContext& renderer,
+                     const Entity& entity,
+                     RenderPass& pass,
+                     const PipelineBuilderCallback& pipeline_callback,
+                     typename VertexShaderT::FrameInfo frame_info,
+                     const BindFragmentCallback& bind_pipeline_callback) const {
+    auto geometry_result =
+        GetGeometry()->GetPositionBuffer(renderer, entity, pass);
+
+    return DrawGeometry<VertexShaderT>(geometry_result, renderer, entity, pass,
```

Comment:
```suggestion
    return DrawGeometry<VertexShaderT>(std::move(geometry_result), renderer, entity, pass,
```

geometry_result contains shared_ptrs in the buffer views, should be moved.

Reply:
Done.